iib/workers/s3_utils.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+import boto3
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+
+log = logging.getLogger(__name__)
+
+
+def upload_file_to_s3_bucket(file_path, key_prefix, file_key):
+    """
+    Upload artifact file to AWS S3 bucket.
+
+    :param str file_path: the path of the file locally where the artifact file is stored
+    :param str key_prefix: the logical location of the file in the S3 bucket
+    :param str file_key: the name of the file in S3 bucket
+    :raises IIBError: when unable to upload file to the S3 bucket
+    """
+    conf = get_worker_config()
+    try:
+        log.info(
+            'Uploading file %s/%s to S3 bucket: %s',
+            key_prefix,
+            file_key,
+            conf['iib_aws_s3_bucket_name'],
+        )
+        s3 = boto3.resource(service_name='s3')
+        s3.meta.client.upload_file(
+            Filename=file_path,
+            Bucket=conf['iib_aws_s3_bucket_name'],
+            Key=f'{key_prefix}/{file_key}',
+        )
+    except Exception as error:
```

Comment:
Do you have somewhere list of Exceptions that can be raised? I did not find one quickly. 
I know that it is not good practice to catch broad Exception.

Reply:
I agree with the point of it not being a good practice but it's a long list so I decided to catch the general exception. Please let me know if you think there's something else I should do

(https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html) Check out "List of Error Codes" section.

Reply:
@yashvardhannanavati what about getting inspiration from this example: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-uploading-files.html#uploading-files 
catching `ClientError` from `botocore.exceptions` might be better then catching general `Exception`.

More on error handling: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#catching-botocore-exceptions  
> The most common botocore exception you’ll encounter is ClientError. This is a general exception when an error response is provided by an AWS service to your Boto3 client’s request.

Reply:
Done